lib/services/audio/default_audio_player_service.dart

```diff
@@ -1053,7 +1057,11 @@ class _DefaultAudioPlayerHandler extends BaseAudioHandler with SeekHandler {
         storedEpisode.position = 0;
         storedEpisode.played = true;
 
-        await repository.saveEpisode(storedEpisode);
+        if (settings.deleteDownloadedPlayedEpisodes) {
+          repository.deleteEpisode(storedEpisode);
```

Comment:
Here you are calling the repository directly to delete the episode. Although this is similar to the save above, when you delete an episode a number of additional actions are performed such as marking the episode as played (if set in settings) and tidying up any associated transcripts. By deleting directly from the repository these steps are missed.

I think it would be better to call the `deleteDownload` in the podcast service instance that is available to the audio player service. If you take a look you can see it carries out several steps before calling `deleteEpisode` on the repository.

Reply:
Hi @amugofjava, I agree. I had not paid attention to the additional actions. I make the correction.